#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "../protos/SimpleVehicle.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/Parquetry.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/living_room_furniture/protos/Sofa.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/geometries/protos/Rectangle.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/plants/protos/PottedTree.proto"
EXTERNPROTO "../protos/RobotisLds01.proto"

WorldInfo {
  basicTimeStep 16
  FPS 15
  contactProperties [
    ContactProperties {
      coulombFriction [
        3
      ]
    }
  ]
}
Viewpoint {
  orientation -0.5773502691896257 0.5773502691896257 0.5773502691896257 2.09
  position -0.03741040480125733 -0.1846339489705111 66.85664677536846
  follow "vehicle:solid"
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 30 30
  floorTileSize 1 1
  floorAppearance Parquetry {
    type "light strip"
  }
  wallHeight 1
}
DEF Wall_Identifiers Group {
  children [
    DEF Turn_1 Group {
      children [
        Solid {
          translation 14.93 12.38 0.51
          rotation 0 0 1 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
              }
              geometry Box {
                size 0.4 0.1 0.1
              }
            }
          ]
        }
        Solid {
          translation 13.04 14.97 0.51
          rotation 0 0 1 -3.1415853071795863
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 1 0
              }
              geometry Box {
                size 0.4 0.1 0.1
              }
            }
          ]
          name "solid(2)"
        }
      ]
    }
    DEF Turn_2 Group {
      children [
        Solid {
          translation 13.04 -14.99 0.51
          rotation 0 0 1 -3.1415853071795863
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
              }
              geometry Box {
                size 0.4 0.1 0.1
              }
            }
          ]
          name "solid(1)"
        }
        Solid {
          translation 14.93 -13.41 0.51
          rotation 0 0 1 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 1 0
              }
              geometry Box {
                size 0.4 0.1 0.1
              }
            }
          ]
          name "solid(3)"
        }
      ]
    }
    DEF Turn_3 Group {
      children [
        Solid {
          translation -13.74 -14.99 0.51
          rotation 0 0 1 -3.1415853071795863
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 1 0
              }
              geometry Box {
                size 0.4 0.1 0.1
              }
            }
          ]
          name "solid(4)"
        }
        Solid {
          translation -14.99 -13.41 0.51
          rotation 0 0 1 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
              }
              geometry Box {
                size 0.4 0.1 0.1
              }
            }
          ]
          name "solid(5)"
        }
      ]
    }
  ]
}
DEF SimpleVehicle SimpleVehicle {
  translation -0.4754 11.57 0.00993884
  rotation -0.6255311561726801 -0.09122411120461252 -0.7748476845110908 0.00287717
  controller "RaceCar_RL_Controller_05_BetterLapTimes_LoadedModel"
  supervisor TRUE
  bodyExpansionSlot [
    Solid {
      translation 0.14 0 0.01
      children [
        DEF platform Shape {
          geometry Box {
            size 0.3 0.14 0.01
          }
        }
      ]
      boundingObject USE platform
      physics Physics {
      }
    }
  ]
  sensorExpansionSlot [
    Transform {
      translation 0.27 0 0.018
      children [
        Accelerometer {
          translation -0.13 0 0.01
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.02 0.02 0.02
              }
            }
          ]
        }
        Transform {
          children [
            Camera {
              translation -0.27 1.59216e-07 0
              rotation 0 0 1 3.14159
              children [
                Shape {
                  appearance PBRAppearance {
                  }
                  geometry Box {
                    size 0.02 0.02 0.02
                  }
                }
              ]
              name "rear_camera"
              width 128
              height 96
            }
            Solid {
              translation -0.06 0 0.02
              children [
                Transform {
                  translation 0 0 0.04
                  children [
                    Camera {
                      translation 0.02 0 0.02
                      children [
                        Shape {
                          appearance PBRAppearance {
                          }
                          geometry Box {
                            size 0.02 0.02 0.02
                          }
                        }
                      ]
                      width 128
                      height 96
                    }
                    RobotisLds01 {
                      translation 0.01 0 0
                      enablePhysics FALSE
                    }
                  ]
                }
                Shape {
                  appearance PBRAppearance {
                  }
                  geometry Box {
                    size 0.15 0.05 0.05
                  }
                }
              ]
            }
          ]
        }
      ]
    }
  ]
  boundingObject Transform {
    translation 0.14 0 0.01
    children [
      USE platform
    ]
  }
}
Transform {
  rotation 0 1 0 0
  scale 1 1 2
  children [
    Solid {
      translation 0 12.44 0.001
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "finish_line.jpg"
              ]
              repeatT FALSE
            }
            textureTransform TextureTransform {
              scale 6.5 1
            }
          }
          geometry Rectangle {
            size 5.1 0.1
          }
        }
      ]
      name "start line"
    }
    Solid {
      translation 0 0 0.1
      rotation 0.577656895078166 -0.5771968951617173 -0.5771968951617173 -2.093935307179586
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "track_id_rectangle.png"
              ]
              repeatS FALSE
              repeatT FALSE
              filtering 0
            }
          }
          geometry Rectangle {
            size 12 6
          }
        }
      ]
      name "course billboard"
    }
    Solid {
      translation 0.57 9.86 0
      children [
        DEF NORTH_WALL Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 20 0.1 1
          }
        }
      ]
      name "north wall"
      boundingObject USE NORTH_WALL
    }
    Solid {
      translation 0.57 -9.57 0
      children [
        DEF SOUTH_WALL Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 20 0.1 1
          }
        }
      ]
      name "south wall"
      boundingObject USE SOUTH_WALL
    }
    Solid {
      translation 10.61 0.88 0
      rotation 0 0 1 1.5708
      children [
        DEF EAST_WALL Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 21 0.1 1
          }
        }
      ]
      name "east wall"
      boundingObject USE EAST_WALL
    }
    Solid {
      translation -9.48 -0.59 0
      rotation 0 0 1 1.5708
      children [
        DEF WEST_WALL Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 21 0.1 1
          }
        }
      ]
      name "west wall"
      boundingObject USE WEST_WALL
    }
  ]
}
Sofa {
  translation 0.87 -12.6 0
  rotation 0 0 1 2.0944
}
Group {
  children [
    PottedTree {
      translation -14.23 14.08 0
    }
    PottedTree {
      translation -14.23 13.5 0
      name "potted tree(1)"
    }
    PottedTree {
      translation -13.1 14.08 0
      name "potted tree(2)"
    }
    PottedTree {
      translation -13.01 13.51 0
      name "potted tree(3)"
    }
  ]
}
